sofl/importer/desktop_source.py

```
@@ -177,14 +177,23 @@ def __iter__(self):
-        flatpak_str = "flatpak-spawn --host /bin/sh -c "
-
-        if os.getenv("FLATPAK_ID") == shared.APP_ID:
-            command = flatpak_str + shlex.quote(command)
-
         try:
-            subprocess.run(command, shell=True, check=True)
-        except subprocess.CalledProcessError:
+            # Безопасная альтернатива shell=True - используем which напрямую
+            if command.startswith("which "):
+                executable_name = command[6:]  # Убираем "which "
+                result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)